lib/clients/apifutebol/api_client.rb

```diff
@@ -1,10 +1,109 @@
 # frozen_string_literal: true
 
+require 'net/http'
+
 # Soccer API Client
 class Client
-  def matches
+  def league
     filename = File.expand_path('matches.json', File.dirname(__FILE__))
     file = File.read(filename)
-    JSON.parse(file)
+    data = JSON.parse(file)
+    ImportLeague.new(data)
+  end
+
+  # https://api.soccerdataapi.com/match/?match_id=[id]&auth_token=[AUTH]
+  def match(id) # rubocop:disable Metrics/AbcSize
+    auth_token = ENV.fetch('SOCCER_AUTH_TOKEN', nil)
+    raise "SOCCER_AUTH_TOKEN is required" if auth_token.nil?
+
+    base_url = 'https://api.soccerdataapi.com/match/'
+
+    uri = URI("#{base_url}?match_id=#{id}&auth_token=#{auth_token}")
+    # puts "URI is #{uri}"
+
+    Net::HTTP.start(uri.host, uri.port,
+                    use_ssl: uri.scheme == 'https') do |http|
+      request = Net::HTTP::Get.new uri
+
+      response = http.request request # Net::HTTPResponse object
+      if response.is_a?(Net::HTTPSuccess)
+        # Parse the JSON response
+        parsed_response = JSON.parse(response.body)
+        return ImportMatch.new(parsed_response)
+
+      else
+        puts "Request failed with HTTP status code: #{response.code}"
+        puts "Response Body: #{response.body}"
```

Comment:
`puts` is fine when debugging dev code, but for production code, you'll want to look into using [`Rails.logger` for this stuff](https://guides.rubyonrails.org/debugging_rails_applications.html#the-logger). There's a [rubocop rule in `rubocop-rails` to prevent you from doing this, too](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsoutputsafety).

Reply:
Migrated to Rails logger.